samcli/lib/sync/flows/image_function_sync_flow.py

```diff
@@ -51,20 +49,21 @@ def __init__(
             Physical ID Mapping
         stacks : Optional[List[Stack]]
             Stacks
-        docker_client : Optional[DockerClient]
-            Docker client to be used for building and uploading images.
-            Defaults to docker.from_env() if None is provided.
         """
         super().__init__(function_identifier, build_context, deploy_context, sync_context, physical_id_mapping, stacks)
         self._ecr_client = None
         self._image_name = None
-        self._docker_client = docker_client
+        self._docker_client = None
 
-    def set_up(self) -> None:
-        super().set_up()
-        self._ecr_client = self._boto_client("ecr")
+    def _get_docker_client(self) -> DockerClient:
         if not self._docker_client:
             self._docker_client = docker.from_env()
+        return self._docker_client
+
+    def _get_ecr_client(self) -> Any:
+        if not self._ecr_client:
+            self._ecr_client = self._boto_client("ecr")
```

Comment:
https://github.com/aws/aws-sam-cli/blob/93462eea9b82d45e386c334490df87a653912594/samcli/lib/utils/boto_utils.py#L15

Should we pass in a config? Also worth noting that we may be missing it in other places.

Reply:
We are already generating a client with config, see call trace below;
- [`self._boto_client`](https://github.com/mndeveci/aws-sam-cli/blob/update_local_hash/samcli/lib/sync/sync_flow.py#L96)
- [`get_boto_client_provider_from_session_with_config`](https://github.com/mndeveci/aws-sam-cli/blob/update_local_hash/samcli/lib/utils/boto_utils.py#L44)

Reply:
NIT: docstring for methods `_get_ecr_client` and `_get_docker_client`